hd-wallet/pluginv2.lua

```diff
@@ -0,0 +1,511 @@
+local PRIVATE_WALLET_VERSION =  "0488ADE4"
+local PUBLIC_WALLET_VERSION = "0488B21E"
+local FIRST_HARDENED_CHILD = 2147483648
+local PUBLIC_KEY_COMPRESSED_LENGTH = 33
+local N = "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141"
+------------- BIP32 key structure -------------
+local key = {["version"]="",
+    ["depth"]="",        -- 1 byte
+    ["index"]="",        -- 4 byte child number
+    ["fingerprint"]="",  -- 4 byte parent fingerprint
+    ["chain_code"]="",   -- 32 byte
+    ["key"]="",          -- 33 byte long key
+    ["checksum"]="",     -- checksum of all above
+    ["is_private"] = ""} -- 1 bit flag
+
+function create_ASN1_private_key(key_byte)
+  while string.len(key_byte) < 64 do
+    key_byte = '00' .. key_byte
+  end
+  return "302E0201010420".. key_byte .."A00706052B8104000A"
+end
+-- return hex of exported key
+function decode_key(exported_master_key_serialized)
+  local blob = Blob.from_base58(exported_master_key_serialized)
+  return blob:hex()
+end
+-- deserialize bip32 key
+function deserialize(exported_master_key_serialized)
+    hex_key = decode_key(exported_master_key_serialized)
+
+    key.version = string.sub(hex_key, 1, 8)
+    key.depth = string.sub(hex_key, 9, 10)
+    key.index = string.sub(hex_key, 11, 18)
+    key.fingerprint = string.sub(hex_key, 19, 26)
+    key.chain_code = string.sub(hex_key, 27, 90)
+    key.key = string.sub(hex_key, 91, 156)
+
+    if key.version == "0488ADE4" or key.version == "0x04358394" then
+        key.is_private = 1
+    else
+        key.is_private = 0
+    end
+
+    key.checksum = string.sub(hex_key, 157, 164)
+    return key
+end
+-- import ec key into sdkms
+-- this helps to evaluate public key from private key
+function import_ec_key(blob)
+    local sobject = assert(Sobject.import { name = "ec", obj_type = "EC", elliptic_curve = "SecP256K1", value = blob, transient = true })
+    return sobject
+end
+-- export secret key that holds the BIP32 master key --
+-- export BIP32 master key from SDKMS ---
+function export_secret_key(keyId)
+    return Sobject { kid = keyId }:export().value
+end
+
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+----------           UTXO            ----------
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+
+---------- integer to bytearry impl -----------
+---------- utils methods ----------------------
+local random = math.random
+-- convert number into hex string 
+function num_2_hex(num, size)
+    local hexstr = '0123456789ABCDEF'
+    local s = ""
+    while num > 0 do
+        local mod = math.fmod(num, 16)
+        s = string.sub(hexstr, mod+1, mod+1) .. s
+        num = math.floor(num / 16)
+    end
+    if (string.len(s) < size) then
+        local offset = string.rep("0", size-string.len(s))
+        s = offset..s
+    end
+    return s
+end
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+------------- Parse Derivation Path -----------
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+function split(str, pat)
```

Comment:
Since it's only used for parse_path, it would make more sense to put this code there.

Reply:
This is only used once, weird to make into a separate function.  Move it into the parse_path function